web/src/test/java/edu/softserve/zoo/rest/request/RequestValidationTest.java

```diff
@@ -0,0 +1,122 @@
+package edu.softserve.zoo.rest.request;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.softserve.zoo.controller.rest.Routes;
+import edu.softserve.zoo.dto.InvalidRequestDto;
+import edu.softserve.zoo.dto.WarehouseDto;
+import edu.softserve.zoo.validation.FieldError;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.util.Locale;
+
+import static org.junit.Assert.assertEquals;
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * This class is used while testing Invalid Request processing. {@link WarehouseDto} was used as an example.
+ *
+ * @author Andrii Abramov on 6/13/16.
+ */
+
+@RunWith(value = SpringJUnit4ClassRunner.class)
+@ContextConfiguration("classpath:spring/web-test-ctx.xml")
+@WebAppConfiguration
+@ActiveProfiles("test")
+public class RequestValidationTest {
+
+    private static final Locale[] availableLocales = {Locale.forLanguageTag("uk"), Locale.ENGLISH};
+
+    @Autowired
+    private MessageSource messageSource;
+
+    @Autowired
+    private WebApplicationContext context;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private MockMvc mockMvc;
+
+
+    @Before
+    public void setUp() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(context)
+                .dispatchOptions(true)
+                .build();
+    }
+
+
+    @Test
+    public void testInvalidPatch() throws Exception {
+
+        final WarehouseDto warehouseDto = getRequestBody();
+        final String body = objectMapper.writeValueAsString(warehouseDto);
+
+        for (final Locale locale : availableLocales) {
+
+            final MockHttpServletRequestBuilder request = patch(Routes.WAREHOUSES + '/' + warehouseDto.getId())
```

Comment:
hm.. these test is not for CRUD logic.
Also I think about writing test if there is custom i18ned validation message for every constraint in dtos.

Okay, I'll try to write those tests.